packages/hooks/use-css-var/index.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  watch,
+  isRef,
+  onMounted,
+  computed,
+  unref,
+  inject,
+  onUnmounted,
+  provide,
+  ref,
+} from 'vue'
+import type { Ref, WatchStopHandle, InjectionKey } from 'vue'
+
+type VarsType = Ref<Record<string, string>> | Record<string, string>
+
+const VAR_PREFIX = '--el-'
+
+const setVars = (
+  target: HTMLElement,
+  val: Record<string, string>,
+) => {
+  Object.keys(val).forEach(key => {
+    if (key.startsWith(VAR_PREFIX)) {
+      target?.style.setProperty(key, val[key])
+    } else {
+      target?.style.setProperty(VAR_PREFIX + key, val[key])
+    }
+  })
+}
+
+export const themeVarsKey: InjectionKey<VarsType> = 'themeVars' as any
+
+/**
+ * @param vars
+ * @param target
+ * @returns stopWatchHandler
+ * @example
+ *
+ * If you pass a Ref vars param, it will generate a watch handler.
+ *
+ * In main.ts:
+ *
+ * ```ts
+ * const themeVars = {
+ *  '--el-color-primary': '#f44336',
+ *  '--el-color-white': '#2196f3',
+ * }
+ * app.provide(themeVarsKey, themeVars)
+ * ```
+ * Usually you need to use this function in the root component.
+ * In components' steup:
+ *
+ * ```ts
+ * setup() {
+ *  const themeVars = ref({
+ *    '--el-button-default-background-color': '#f44336',
+ *    '--el-button-default-font-color': '#2196f3',
+ *  })
+ *  useCssVar(themeVars)
+ * }
+ * ```
+ */
+export function useCssVar(
+  vars: VarsType,
+  target?: Ref<HTMLElement> | HTMLElement,
+) {
+  let stopWatchCssVar: WatchStopHandle | null = null
```

Comment:
I more thing, we should do something like this:

```ts
import isServer from '@element-plus/util/isServer`
setup() {
    if (isServer) return
}
```

Reply:
Since this whole css thing would work without actual DOM elements, so we can say that this judgement is legit.

Reply:
`optional chaining` prevents this error from occurring

Reply:
I understand that but isn't early terminating better than optional chaining? Besides, even though we use optional chaining throughout the project I think this is a better way to do that instead of relying on optional chaining (Plus, deep optional chaining gets compiled into ugly code even though we don't have to read that code but it definitely increases the bundle size with `window ? window.document ? window.document.documentElement : null : null : null`